### PR TITLE
Use different thread pool for connections in dbTransactor.

### PR DESCRIPTION
**Pull request by irumiha**
Don't know why but it seems to solve #149.

### CODE REVIEW

File: src/main/scala/io/github/pauljamescleary/petstore/Server.scala [hunk 1]
```diff
@@ -12,14 +12,16 @@ import org.http4s.server.{Server => H4Server, Router}
...
+import doobie.util.ExecutionContexts
 import scala.concurrent.ExecutionContext.Implicits.global
 import io.circe.config.parser
 
 object Server extends IOApp {
   def createServer[F[_] : ContextShift : ConcurrentEffect : Timer]: Resource[F, H4Server[F]] =
     for {
       conf           <- Resource.liftF(parser.decodePathF[F, PetStoreConfig]("petstore"))
-      xa             <- DatabaseConfig.dbTransactor(conf.db, global, global)
+      connEc         <- ExecutionContexts.fixedThreadPool[F](10)
```
Comment (zakpatterson) [inline src/main/scala/io/github/pauljamescleary/petstore/Server.scala]: This is a great catch. Do you think it's worth adding a configuration option for the number of connection threads, so we don't have that `10` or some other number in there?

Also, we should probably follow the documentation on connections, here, which your code agrees with, but should also fix/understand the use of global on the following line.
https://tpolecat.github.io/doobie/docs/14-Managing-Connections.html#using-a-hikaricp-connection-pool

Comment (irumiha) [reply src/main/scala/io/github/pauljamescleary/petstore/Server.scala]: Re: configuration option - yes, I'll amend the PR.

Actually the suggested use of the unbounded thread pool for transaction execution should also go here. If you agree I'll add that too.

Comment (zakpatterson) [reply src/main/scala/io/github/pauljamescleary/petstore/Server.scala]: Yep I think fixing up both and adding the configuration option would make this great.

Comment (pauljamescleary) [reply src/main/scala/io/github/pauljamescleary/petstore/Server.scala]: I think just upping the `maxConnectionPoolSize` setting on the hikari CP should do the trick.

I do agree that the separate execution context is the right thing as well.

There should be a way to call `Executors.xxx` and get the right sized thread pool.  it should be based by the number of available processors.

**Comment by zakpatterson** [comment]
I think this is the right correction, @paulcleary give a 👍 since you weighed in on the comments above

**Pull request merged**